rostok/simulation_chrono/simulation_utils.py

```diff
@@ -29,3 +54,82 @@ def set_covering_sphere_based_position(obj: ChronoEasyShapeObject,
+                                              0, 0, 0),
+                                          direction: chrono.ChVectorD = chrono.ChVectorD(0, 1, 0)):
+    center, axis = calculate_covering_ellipsoid(obj)
+    direction.Normalize()
+    desired_position = reference_point + direction * axis[1] / 2
+    shift = desired_position - obj.body.GetCoord().TransformPointLocalToParent(center)
+    current_cog_pos = obj.body.GetPos()
+    obj.body.SetPos(current_cog_pos + shift)
+
+
+def set_covering_ellipsoid_based_position(obj: ChronoEasyShapeObject,
+                                          reference_point: chrono.ChVectorD = chrono.ChVectorD(
+                                              0, 0, 0),
+                                          direction: chrono.ChVectorD = chrono.ChVectorD(0, 1, 0)):
+    center, axis = calculate_covering_ellipsoid(obj)
+    direction.Normalize()
+    desired_position = reference_point + direction * axis[1] / 2
+    shift = desired_position - obj.body.GetCoord().TransformPointLocalToParent(center)
+    current_cog_pos = obj.body.GetPos()
+    obj.body.SetPos(current_cog_pos + shift)
+
+
+@dataclass
+class SimulationResult:
+    """Data class to aggregate the output of the simulation.
+    
+        Attributes:
+            time (float): the total simulation time
+            time_vector (List[float]): the vector of time steps
+            n_steps (int): the maximum possible number of steps
+            robot_final_ds (Optional[DataStorage]): final data store of the robot
+            environment_final_ds (Optional[DataStorage]): final data store of the environment"""
+    time: float = 0
+    time_vector: List[float] = field(default_factory=list)
+    n_steps = 0
+    robot_final_ds: Optional[DataStorage] = None
+    environment_final_ds: Optional[DataStorage] = None
+
+    def reduce_ending(self, step_n):
+        if self.robot_final_ds:
+            storage = self.robot_final_ds.main_storage
+            for key in storage:
+                key_storage = storage[key]
+                for key_2 in key_storage:
+                    value = key_storage[key_2]
+                    new_value = value[:step_n + 2:]
+                    key_storage[key_2] = new_value
+
+        if self.environment_final_ds:
+            storage = self.environment_final_ds.main_storage
+            for key in storage:
+                key_storage = storage[key]
+                for key_2 in key_storage:
+                    value = key_storage[key_2]
+                    new_value = value[:step_n + 2:]
+                    key_storage[key_2] = new_value
+
+    def reduce_nan(self):
+        if self.robot_final_ds:
+            storage = self.robot_final_ds.main_storage
+            for key in storage:
+                key_storage = storage[key]
+                for key_2 in key_storage:
+                    value = key_storage[key_2]
+                    new_value = [x for x in value if np.logical_not(np.isnan(x).all())]
```

Comment:
Как-то сложно. Заменить 

```suggestion
                    new_value = x = x[~numpy.isnan(x)]
```

Reply:
Можно потом в лист преобразовать